[Pages] New domain limits (#5807)
cloudflare/cloudflare-docs: content/pages/platform/limits.md

## Custom domains

> A Cloudflare Pages project can be attached to a certain number of domains at a time.

Reviewer: ```suggestion
A Cloudflare Pages project can be attached to a certain number of domains per plan.
```
